File: app/src/main/java/com/example/superimagesviewer/RecyclerAdapter.java
package com.example.superimagesviewer;

import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ImageViewHolder> {

    private List<Drawable> drawables = new ArrayList<>();

    protected static class ImageViewHolder extends RecyclerView.ViewHolder {

        ImageView photo;

        ImageViewHolder(View itemView) {
            super(itemView);
            photo = itemView.findViewById(R.id.photo);
        }
    }

    @Override
    public ImageViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.mosaic_item, parent, false);

        return new ImageViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ImageViewHolder holder, int position) {
        holder.photo.setImageDrawable(drawables.get(position));
    }

    public void setDrawables(List<Drawable> drawables) {
        this.drawables = drawables;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return drawables.size();
    }

}

File: app/src/main/java/com/example/superimagesviewer/RecyclerAdapter.kt
package com.example.superimagesviewer

import android.graphics.drawable.Drawable
import androidx.recyclerview.widget.RecyclerView
import com.example.superimagesviewer.RecyclerAdapter.ImageViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import java.util.ArrayList

class RecyclerAdapter : RecyclerView.Adapter<ImageViewHolder>() {
    private var drawables: List<Drawable> = ArrayList()

    class ImageViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var photo: ImageView = itemView.findViewById(R.id.photo)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.mosaic_item, parent, false)
        return ImageViewHolder(view)
    }

    override fun onBindViewHolder(holder: ImageViewHolder, position: Int) {
        holder.photo.setImageDrawable(drawables[position])
    }

    fun setDrawables(drawables: List<Drawable>) {
        this.drawables = drawables
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return drawables.size
    }

}
